internal/ls/organizeimports.go

```diff
@@ -126,3 +127,287 @@ func isIndexFileName(fileName string) bool {
+		}
+		importClause := importDecl.ImportClause.AsImportClause()
+		if importClause.IsTypeOnly() {
+			return 1 // Type-only import
+		}
+		if importClause.NamedBindings != nil && importClause.NamedBindings.Kind == ast.KindNamespaceImport {
+			return 2 // Namespace import
+		}
+		if importClause.Name() != nil {
+			return 3 // Default import
+		}
+		return 4 // Named imports
+	case ast.KindImportEqualsDeclaration:
+		return 5
+	case ast.KindVariableStatement:
+		return 6 // Require statement
+	default:
+		return 7
+	}
+}
+
+// compareImportsOrRequireStatements compares two import or require statements for sorting
+func compareImportsOrRequireStatements(s1 *ast.Statement, s2 *ast.Statement, comparer func(a, b string) stringutil.Comparison) stringutil.Comparison {
+	if cmp := compareModuleSpecifiersWorker(getModuleSpecifierExpression(s1), getModuleSpecifierExpression(s2), comparer); cmp != stringutil.ComparisonEqual {
+		return cmp
+	}
+	return compareImportKind(s1, s2)
+}
+
+// compareImportOrExportSpecifiers compares two import or export specifiers
+func compareImportOrExportSpecifiers(s1 *ast.Node, s2 *ast.Node, comparer func(a, b string) stringutil.Comparison, preferences *UserPreferences) stringutil.Comparison {
+	typeOrder := OrganizeImportsTypeOrderLast
+	if preferences != nil {
+		typeOrder = preferences.OrganizeImportsTypeOrder
+	}
+
+	s1Name := s1.Name().Text()
+	s2Name := s2.Name().Text()
+
+	switch typeOrder {
+	case OrganizeImportsTypeOrderFirst:
+		if cmp := compareBooleans(s2.IsTypeOnly(), s1.IsTypeOnly()); cmp != 0 {
+			return cmp
+		}
+		return comparer(s1Name, s2Name)
+	case OrganizeImportsTypeOrderInline:
+		return comparer(s1Name, s2Name)
+	default: // OrganizeImportsTypeOrderLast
+		if cmp := compareBooleans(s1.IsTypeOnly(), s2.IsTypeOnly()); cmp != 0 {
+			return cmp
+		}
+		return comparer(s1Name, s2Name)
+	}
+}
+
+// getNamedImportSpecifierComparer returns a comparer function for import/export specifiers
+func getNamedImportSpecifierComparer(preferences *UserPreferences, comparer func(a, b string) stringutil.Comparison) func(s1, s2 *ast.Node) stringutil.Comparison {
+	if comparer == nil {
+		ignoreCase := false
+		if preferences != nil && !preferences.OrganizeImportsIgnoreCase.IsUnknown() {
+			ignoreCase = preferences.OrganizeImportsIgnoreCase.IsTrue()
+		}
+		comparer = getOrganizeImportsOrdinalStringComparer(ignoreCase)
+	}
+	return func(s1, s2 *ast.Node) stringutil.Comparison {
+		return compareImportOrExportSpecifiers(s1, s2, comparer, preferences)
+	}
+}
+
+// getImportSpecifierInsertionIndex finds the insertion index for a new import specifier
+func getImportSpecifierInsertionIndex(sortedImports []*ast.Node, newImport *ast.Node, comparer func(s1, s2 *ast.Node) stringutil.Comparison) int {
+	if len(sortedImports) == 0 {
+		return 0
+	}
+
+	low := 0
+	high := len(sortedImports) - 1
+
+	for low <= high {
+		middle := low + ((high - low) >> 1)
+		midKey := sortedImports[middle]
+		switch comparer(midKey, newImport) {
```

Comment:
I think this is a pure port, but I am always suspicious of code which exactly checks these comparison constants.